### ISSUE TITLE
Order of Compare-selected and Compare-with-selected

### ISSUE THREAD

**Issue by lijh8** [microsoft/vscode#226883]
Does this issue occur when all extensions are disabled?: Yes/No

- VS Code Version: 1.92.2 (Universal)
- OS Version: macOS Version: 12.7.6 (21H1320)

Steps to Reproduce:

1. 

Compare-selected by mouse from Folders in explorer window,
the first selected file appears on the right in diff window

This is different from: 
1. using the menu: select for compare, compare with selected;
2. Compare-selected from Open Editors in explorer window;
3. Compare-selected from tab;

2.

Compare-with-selected can not select the correct files for diff 
in Folders and Open Editors, 
if the filename appears first alphabetically is selected first.

**Comment by lramos15** [comment]
Sorry I'm not understanding the issue. Could you try rephrasing it a bit so that I may reproduce?

**Comment by Namarot** [comment]
@lramos15 
I was going to open a closely related issue, but I will elaborate further here, let me know if I should open a new issue instead.

The behavior @lijh8 describes in step 1 has now been reversed in Version 1.93.0, see: #225441 
However the behavior of "Compare Selected" is still inconsistent between "Folders" and "Open Editors".

- VS Code Version: 
Version: 1.93.0
Commit: 4849ca9bdf9666755eb463db297b69e5385090e3
Date: 2024-09-04T13:02:38.431Z
Electron: 30.4.0
ElectronBuildId: 10073054
Chromium: 124.0.6367.243
Node.js: 20.15.1
V8: 12.4.254.20-electron.0
OS: Darwin arm64 23.6.0
- OS Version: macOS Sonoma Version 14.6.1 (23G93)

Steps to Reproduce:
- Click on a file in the "Folders" view of the Explorer to select it
- Command + click on another file in the "Folders" view of the Explorer to select that one as well
- Right click on either file and click "Compare Selected"
- Now repeat the previous steps for 2 files in the "Open Editors" view of the Explorer (note which file you right click)

Actual Behavior:
- In the "Folders" view, the first file is always shown on the left and the second file is always shown on the right. Which file you right click -> Compare Selected does not matter.
- In the "Open Editors" view, the order in which you select the files do not matter. The file you right click -> Compare Selected is always shown on the right.

Expected Behavior:
Right Click -> Compare Selected should have the same behavior in "Folders" and "Open Editors" views.

**Comment by runxel** [comment]
Yes, I second that the whole UX around the Compare feature is utterly borked, see like #232841.
Not only that it _seems_ totally random for the user with what they will end up with, it is also highly problematic and _will_ lead to data loss inevitably. (I just experienced that)
**Furthermore this issue is accelerated by the fact that it is IMPOSSIBLE to deduce what file is actually on the right or the left!**
Hey folks, please! We need some kind of differentation here!:
![Image](https://github.com/user-attachments/assets/ae600296-629f-4c91-b16d-a565f425f5ba)

The breadcrumbs on the left NEVER changes. But what does it mean? Just put two breadcrumbs in there for left and right and it will much clearer.

Also regarding the standard order: If the user chooses two files for compare, the one on the right should always automatically be the _older_ one (because it works from left to right, and I most possibly want newer stuff to come in the older file; like standard git workflow. Or make a switch for that, so people can choose).
